---
title: C6318 |Microsoft Docs
ms.custom: ''
ms.date: 2018-06-30
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C6318
helpviewer_keywords:
- C6318
ms.assetid: 3284a83e-bb8e-461c-adcc-cfc66ceea05e
caps.latest.revision: 15
author: corob-msft
ms.author: gewarren
manager: ghogen
ms.openlocfilehash: 8b737a36066861704e27fa2ce37a980d4674518a
ms.sourcegitcommit: 55f7ce2d5d2e458e35c45787f1935b237ee5c9f8
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 08/22/2018
ms.locfileid: "47478442"
---
# <a name="c6318"></a>C6318
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

本主题的最新版本，请参阅[C6318](https://docs.microsoft.com/visualstudio/code-quality/c6318)。  
  
警告 C6318： 未正确定义 __try /\__except： 使用了常量 EXCEPTION_CONTINUE_SEARCH 或另一个常量，它的计算结果为零异常筛选器表达式中。 不执行异常处理程序块中的代码  
  
 此警告意味着，是否此结构化的异常处理程序的受保护块中发生异常，处理该异常将不会因为该常量`EXCECPTION_CONTINUE_SEARCH`异常筛选器表达式中使用。  
  
 此代码是等效于受保护的块，而无需异常处理程序块，因为不执行处理程序块。  
  
## <a name="example"></a>示例  
 下面的代码生成此警告：  
  
```  
#include <excpt.h>  
#include <stdio.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except (EXCEPTION_CONTINUE_SEARCH) // warning 6318  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 若要更正此警告，请使用以下代码：  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except( (GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION) ?  
                EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )  
  {  
    puts("Access violation");  
  }  
}   
```  
  
## <a name="see-also"></a>请参阅  
 [try-except 语句](http://msdn.microsoft.com/library/30d60071-ea49-4bfb-a8e6-7a420de66381)



